core/workspace.js

```diff
@@ -131,6 +131,15 @@ Blockly.Workspace.prototype.MAX_UNDO = 1024;
  */
 Blockly.Workspace.prototype.connectionDBList = null;
 
+/**
+ * Refresh workspace after changing some blolcks.
+ */
+Blockly.Workspace.prototype.refresh = function() {
+  this.clear();
+  Blockly.Xml.domToWorkspace(Blockly.Xml.workspaceToDom(this), this);
```

Comment:
`workspaceToDom` after the workspace is cleared won't show anything. I believe what you're after is this: 
```
var xml = Blockly.Xml.workspaceToDom(this);
this.clear();
Blockly.Xml.domToWorkspace(xml, this);
```

Also, we need to handle the refresh of the trashcan if it is currently open.

@rachel-fenichel this seems reasonable to me, and is similar to what I'm doing in the new playground to refresh the workspace.
I wonder though, if there's a reason this functionality hadn't been added before, is there some piece of context we're missing here?

Reply:
We didn't have this function because clearing and reloading the workspace with the exact same blocks is almost never the correct thing to do (unless you're changing something core like whether there's a toolbox), and it's expensive.

If you need to rerender all of the blocks, you can get the list of blocks and call render on every one.

If you want to update a bunch of dropdowns, you can get the blocks by type and then update them.

If you want dynamic dropdowns, you can do that with the built-in dynamic dropdown functionality.

And so on.

In this [case](https://github.com/google/blockly/issues/3940), the goal appears to be to do dynamic dropdowns. Redefining the block is an extremely heavy-handed way to accomplish this, and the library provides several other ways to do it, including dynamic dropdowns and mutators.

Reply:
Thanks for the context @rachel-fenichel. 
I agree that, although this appears harmless, it encourages behaviour that yields to poor performance. 
Happy to add it when we get a case where there isn't a better alternative, but for now, with all the cases we know of listed above, there shouldn't be a reason for clearing all blocks and re-adding them.

Reply:
@rachel-fenichel i tried as you suggested before thinking about workspace.refresh() 
Here is an example when i get blocks by id and try to call render on it and nothing happened 
![RTWcUxbflM](https://user-images.githubusercontent.com/30471072/84197547-c04d7a80-aabf-11ea-83bf-9ebece0f551e.gif)

Reply:
By using `workspace.refresh()` we don't need to worry about anything like getting blocks by types and changing them and then try to rerender those blocks in toolbox, workspace's area and trashcan also.

Reply:
Have you tried using a function to populate the list of options in the `FieldDropdown`, also known as dynamic dropdowns: 
https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/dropdown#dynamic_dropdowns

Reply:
actually i want full control of it i want to change full appearance of blocks. I want to update those blocks dynamically may be need to change color or may be want to change full functionality just need to redefine blocks.

Reply:
https://developers.google.com/blockly/guides/create-custom-blocks/web/mutators

Mutators give you even deeper control over your blocks.

If you want this functionality instead of using any of the other ways the library can help you out, you should implement it in your own app, since the functions you need to call are public.

Reply:
thanku
i will try it.